docs_src/design/adr/system-management/0006-Metrics-Collection.md

```diff
@@ -0,0 +1,390 @@
+    ```
+
+!!! Info
+    The key or metric name must be unique when using go-metrics as it requires the metric name to be unique per the registry.  Metrics are considered immutable.
+
+#### Configuration
+
+Configuration, not unlike that provided in core data or any device service, will specify the message bus type and locations where the metrics messages should be sent.  In fact, the message bus configuration will use (or reuse if the service is already using the message bus) the common message bus configuration as defined below.
+
+**Common configuration** for each service for message queue configuration (inclusive of metrics):
+
+``` yaml
+[MessageQueue]
+Protocol = 'redis'  ## or 'tcp'
+Host = 'localhost'
+Port = 5573
+Type = 'redis'  ## or 'mqtt'
+PublishTopicPrefix = "edgex/events/core" # standard and existing core or device topic for publishing  
+  [MessageQueue.Optional]
+  # Default MQTT Specific options that need to be here to enable environment variable overrides of them
+  # Client Identifiers
+  ClientId = "device-virtual"
+  # Connection information
+  Qos = "0" # Quality of Service values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+  KeepAlive = "10" # Seconds (must be 2 or greater)
+  Retained = "false"
+  AutoReconnect = "true"
+  ConnectTimeout = "5" # Seconds
+  SkipCertVerify = "false" # Only used if Cert/Key file or Cert/Key PEMblock are specified
+```
+
+Additional configuration must be provided in each service to provide metrics / telemetry specific configuration.  This area of the configuration will likely be different for each type of service.
+
+**Additional metrics collection configuration** to be provided include: 
+
+- Trigger the collection of telemetry from the metrics cache and sending it into the appointed message bus.
+- Define which metrics are available and which are turned `off` and `on`.  All are false by default.  The list of metrics can and likely will be different per service.  The keys in this list are the metric name.  True and false are used for `on` and `off` values.
+- Specify the metrics topic prefix where metrics data will be published to (ex:  providing the prefix /edgex/telemetry/topic name where the service and metric name `[service-name]/[metric-name]` will be appended per metric (allowing subscribers to filter by service or metric name)
+
+These metrics configuration options will be defined in the `Writable` area of `configuration.toml` so as to allow for dynamic changes to the configuration (when using Consul).  Specifically, the `[Writable].[Writable.Telemetry]` area will dictate metrics collection configuration like this:
+
+``` yaml
+[[Writable]]
+    [[Writable.Telemetry]]
+    Interval = "30s"
+    TelemetryPublishTopicPrefix  = "edgex/telemetry" # /<service-name>/<metric-name> will be added to this Publish Topic prefix
+    Metrics = "{\"service-up\":\"false\", \"api-requests\":\"false\"}"  #all metrics will be off (or false) by default configuration

```

Comment:
This [[ is Array syntax for TOML. List of Metrics should be a `map` not embedded JSON. ```suggestion
[Writable]
   [Writable.Telemetry]
   Interval = "30s"
   PublishTopicPrefix  = "edgex/telemetry" # /<service-name>/<metric-name> will be added to this Publish Topic prefix
   [Writable.Telemetry.Metrics ]
   service-up = "false"
   api-requests = "false"  #all metrics will be off (or false) by default configuration
```

Reply:
changed accordingly